### PR TITLE
Add `es-x/no-regexp-duplicate-named-capturing-groups` rule

### LINKED ISSUE #142: Add rule for RegExp duplicate named capture groups (ES2025)
https://github.com/tc39/proposal-duplicate-named-capturing-groups

### PR DESCRIPTION
**Pull request by ota-meshi**
close #142

### CODE REVIEW

**Comment by aladdin-add** [comment]
Is it a draft as the proposal has not reached stage-4?

**Comment by ota-meshi** [comment]
First, I am waiting for the release of acorn. It is implemented in the master branch.
https://github.com/acornjs/acorn/commits/master/

After that, we will need to change espree and regexpp.
A PR has been opened for regexpp, but if the source code of acorn changes before the release, we may make changes accordingly.
https://github.com/eslint-community/regexpp/pull/195

**Comment by aladdin-add** [comment]
the supports was added:
* acorn v8.12.0: https://github.com/acornjs/acorn/releases/tag/8.12.0
* regexpp v4.11.0: https://github.com/eslint-community/regexpp/releases/tag/v4.11.0

File: tests/tester.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const RuleTester = getRuleTester()
 const eslintVersion = new Linter().version
 const ecmaVersion =
     /*eslint-disable prettier/prettier */
+    semver.gte(eslintVersion, "9.0.0") ? 2025 :
```
Comment (aladdin-add) [inline tests/tester.js:16]: https://github.com/eslint/eslint/commit/8824aa1469ffc572c5e5c1765d1b6da113dfba19 has not been released, I think it's fine to update to the upcoming version. :)
```suggestion
    semver.gte(eslintVersion, "9.6.0") ? 2025 :
```

**Review by aladdin-add — APPROVED** [review]

**Pull request merged**